java_gen/templates/custom/interface/OFMessage.java

```diff
@@ -0,0 +1,15 @@
+    
+    /**
+     * Compares the two messages for equality, ignoring the XID field.
+     *
+     * @param obj the other message to compare
+     * @return true if the messages are equal, ignoring the XID; false otherwise
+     */
+    boolean equalsIgnoreXid(Object obj);
+
+    /**
+     * Computes the hashcode of the message, ignoring the XID field.
+     *
+     * @return the hashcode of the message, ignoring the XID
+     */
+    int hashCodeIgnoreXid();
```

Comment:
Thanks for clarifying - this makes sense.

Maybe it would be good to incorporate some of this info into the Java-doc?